data-types/datetime64/README.md

```diff
@@ -0,0 +1,33 @@
+# Datetime64 data type
+
+Defines a data type for a datetime object based on a 64-bit integer.
+
+## Permitted fill values
+
+The value of the `fill_value` metadata key must be a signed 64-bit integer.
+
+## Example
+
+For example, the array metadata below specifies that the array uses the datetime64 data type:
+
+```json
+{
+    "data_type": "datetime64",
+    "fill_value": -9223372036854775808,
+    "configuration": {
+      "unit": "s"
+    },
+}
+```
+
+## Notes
+
+Valid values for the `unit` configuration option include: `["Y", "M", "W", "D", "h", "m", "s", "ms", "us", "μs", "ns", "ps", "fs", "as"]`
```

Comment:
Note that the units listed after nanoseconds cannot represent dates beyond 1970 within 64-bits and therefore are presumably not very useful.

Reply:
The first part of this sentence is correct. 

The second is not. There is a lot of use cases in the (paleo-)climate space for dates that go much further back in time.

Reply:
To be clear, the range of 64-bit picoseconds is just a few months forward or backward from the epoch.

Presumably picoseconds would only be useful either with more than 64 bits or for a time duration rather than time point.

Reply:
is it helpful if we define the goal of this datetime type as "give numpy `datetime64` users a zarr data type they can use without thinking"? In this case, we optimize for compatibility and accept whatever decisions the numpy folks have made. We can always define a separate datetime data type that avoids perceived defects in the numpy datetime dtype, and we can point users who don't need maximal numpy compatibility to the better data type.

Reply:
Another comment regarding datetime: numpy supports a special NaT (not a time) value.  Is that intended to be supported here?

Reply:
i'm asking because, whether it's sensible or not, [`ps` is a unit numpy `datetime64` supports](https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime-units). Instead of litigating that decision, or any other numpy decision, we could optimize for compatibility here, which would simplify the task of this PR considerably.

Reply:
> Another comment regarding datetime: numpy supports a special NaT (not a time) value. Is that intended to be supported here?

yes i believe that should be supported, and so it should be defined in this PR

Reply:
Those units don't hurt anything they just are very unlikely to be used intentionally.  Numpy supports the same units for both datetime and timedelta --- are you intending to also add timedelta as well?

Reply:
Those units don't hurt anything they just are very unlikely to be used intentionally.  Numpy supports the same units for both datetime and timedelta --- are you intending to also add timedelta as well?

Reply:
> Those units don't hurt anything they just are very unlikely to be used intentionally. Numpy supports the same units for both datetime and timedelta --- are you intending to also add timedelta as well?

yes, timedelta is planned.